File: src/types/password.ts
export interface PasswordStatus {
  passwordSet: boolean;
  recoveryCodeSet: boolean;
  migrationRequired: boolean;
}

export interface PasswordStrength {
  score: number;
  passed: boolean;
  messages: string[];
}

export interface SetupPasswordResult {
  status: PasswordStatus;
  recoveryCode: string;
}

export interface PasswordActionResult {
  success: boolean;
  message: string;
  strength: PasswordStrength | null;
}

File: src-tauri/src/models/dto/password.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct PasswordStatus {
    pub password_set: bool,
    pub recovery_code_set: bool,
    pub migration_required: bool,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct PasswordStrength {
    pub score: u8,
    pub passed: bool,
    pub messages: Vec<String>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct SetupPasswordResult {
    pub status: PasswordStatus,
    pub recovery_code: String,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct PasswordActionResult {
    pub success: bool,
    pub message: String,
    pub strength: Option<PasswordStrength>,
}
